### PR TITLE
#9492: Change to ttnn matmul in tests and tt_eager

### PR DESCRIPTION
**Pull request by bbradelTT**
### Ticket
Link to Github Issue https://github.com/tenstorrent/tt-metal/issues/9492

### Problem description
- we are porting ops to ttnn

### What's changed
- change matmul calls to point to ttnn matmul in tests and tt_eager
- remove other matmuls from tt_lib since they are no longer used

### Checklist
- [x] Post commit CI passes
- [x] Model regression CI testing passes (if applicable)
- [x] New/Existing tests provide coverage for changes

https://github.com/tenstorrent/tt-metal/actions/runs/9798121182 All post commits passes
https://github.com/tenstorrent/tt-metal/actions/runs/9786220572 Model perf passes
https://github.com/tenstorrent/tt-metal/actions/runs/9785682511 T3k frequent passes
https://github.com/tenstorrent/tt-metal/actions/runs/9785685049 post commit models passes
https://github.com/tenstorrent/tt-metal/actions/runs/9786223069 T3k model perf passes
https://github.com/tenstorrent/tt-metal/actions/runs/9786224832 T3k profiler passes
https://github.com/tenstorrent/tt-metal/actions/runs/9786264210 Nightly fails with the same issues as main and other branches
https://github.com/tenstorrent/tt-metal/actions/runs/9785686817 Device perf fails with the same issues as main
Some tests, especially the T3k ones, were run earlier before the final fine tuning related to specific tests.

### CODE REVIEW

**Review by TT-BrianLiu — APPROVED** [review]

File: tests/tt_eager/profiling/reference.txt [hunk 1]
```diff
@@ -256,7 +255,6 @@ ttnn.lt,40,0.9,1.19,1.24,1.12
 tt_lib.tensor.ltz,40,0.81,0.9,1.12,0.93
 tt_lib.tensor.mac,40,1.69,1.79,2.13,1.83
 tt_lib.tensor.maeloss,40,11.89,13.2,14.13,12.75
-tt_lib.tensor.matmul,40,0.88,1.78,3.89,1.5
```
Comment (ayerofieiev-tt) [inline tests/tt_eager/profiling/reference.txt:259]: Are you sure you want to completely remove the reference for matmul?

**Review by ayerofieiev-tt — APPROVED** [review]
A lot of code removed. We just don’t care about it anymore, was it a dead/deprecated code anyway or?

Some seem to be related to perf testing

File: tests/tt_eager/profiling/reference.txt (hunk 1 above)
Comment (bbradelTT) [reply tests/tt_eager/profiling/reference.txt:259]: Yes.
This is for the python tt_lib operation that I am removing. I don't want any profiling tests trying to run a non-existent op.

**Comment by bbradelTT** [comment]
> A lot of code removed. We just don’t care about it anymore, was it a dead/deprecated code anyway or?
> 
> Some seem to be related to perf testing


With previous PRs, I moved all of the python matmul related calls to exclusively use ttnn.matmul/linear

Therefore all of the tt_lib python paths are now dead and deprecated. I need to remove them before someone starts using them again.

**Comment by bbradelTT** [comment]
@mywoodstock I had to remove the 48x48 conv tests.

They were not using matmul/linear properly because K did not match in the two inputs.

E.g. `tests/tt_eager/python_api_testing/unit_testing/misc/test_moreh_conv.py::test_moreh_conv[math_fidelity0-activations_dtype0-weights_dtype0-2-48-48-160-160-1-1-1-1-0-0-0-True-None-device_params0]` failed because the input shapes were `[1, 1, 51200, 48[64]]` and `[1, 1, 64, 64]`
48 is not equal to 64, therefore this should have failed.

However, the verification in bmm_op was incorrectly checking the padded values instead of the non-padded values.

I think the reason for the second input, the weights, being of dimension 64 instead of 48 is that the weights are padded to the nearest tile, probably via
```
            # for conv op, pad the weights to block shape
            if self.is_1d_systolic:
                weight_tiled_ = ttl.tensor.convert_conv_weight_tensor_to_special_padding_tiled_layout(
                    weight_untiled,
                    weight_block_h_ntiles,
                    weight_block_w_ntiles,
                    output_dtype=weights_dtype,
                )
            else:
                weight_tiled_ = ttl.tensor.convert_conv_weight_tensor_to_tiled_layout(
                    weight_untiled,
                    weight_block_h_ntiles,
                    weight_block_w_ntiles,
                    output_dtype=weights_dtype,
                )
```
in tt_eager/tensor/tensor_utils.cpp

It'd be good if either neither of the inputs are padded or both of the inputs are padded. However, it'd be better if someone who's more familiar with conv does that in a separate PR and then puts the 48x48 tests back in.

File: tests/tt_eager/python_api_testing/unit_testing/misc/test_moreh_conv.py [hunk 2]
```diff
@@ -199,8 +198,6 @@
         # (2, 48, 3, 640, 640, 6, 6, 2, 2, 2, 2, 0, False, None),
         (2, 96, 48, 320, 320, 3, 3, 2, 2, 1, 1, 0, True, {"act_block_h": 32}),
         (2, 48, 96, 160, 160, 1, 1, 1, 1, 0, 0, 0, True, None),
-        (2, 48, 48, 160, 160, 1, 1, 1, 1, 0, 0, 0, True, None),
```
[the remark this answers is not in the document]
Comment (bbradelTT) [reply tests/tt_eager/python_api_testing/unit_testing/misc/test_moreh_conv.py:202]: The test cases are wrong. We weren't checking the bounds properly.

See an earlier comment: https://github.com/tenstorrent/tt-metal/pull/9928#issuecomment-2209506744

I'll comment them out with a remark.

**Review by sankarmanoj-tt — APPROVED** [review]

**Pull request merged**